code/android-core-library/src/main/java/com/adobe/marketing/mobile/utils/remotedownload/RemoteDownloader.kt

```diff
@@ -0,0 +1,94 @@
+/*
+  Copyright 2022 Adobe. All rights reserved.
+  This file is licensed to you under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License. You may obtain a copy
+  of the License at http://www.apache.org/licenses/LICENSE-2.0
+  Unless required by applicable law or agreed to in writing, software distributed under
+  the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+  OF ANY KIND, either express or implied. See the License for the specific language
+  governing permissions and limitations under the License.
+ */
+
+package com.adobe.marketing.mobile.utils.remotedownload
+
+import com.adobe.marketing.mobile.LoggingMode
+import com.adobe.marketing.mobile.MobileCore
+import com.adobe.marketing.mobile.internal.utility.StringUtils
+import com.adobe.marketing.mobile.services.CacheFileService
+import com.adobe.marketing.mobile.services.Networking
+
+/**
+ * Facilitates downloading content from a URL into a local cache directory.
+ * Stores downloaded content in to a directory governed by [CacheFileService].
+ *
+ * Currently, triggers to [download] are independent, successive calls.
+ * This class is intentionally final and components are required to use this via composition;
+ * This allows easier path for components using this class to add queuing logic for jobs internally,
+ * maintaining compatibility when this class adds queuing logic in the future, and facilitating
+ * cleaner tests in composing components.
+ *
+ * TODO: Refactor this class to either eliminate default caching or make caching optional.
```

Comment:
I would prefer moving this into services/util package later. 
We can discuss when implementing cache service and do it as part of that task.